### PR TITLE
Add support for v2.3

### PR DESCRIPTION
**Pull request by jewertow**
I tested SMCP v2.3 running bookinfo and it looks that everything works fine except two minor errors
in the container `install-cni-v2-3`:
```
error   Unable to start ControlZ: listen tcp 127.0.0.1:9876: bind: address already in use
error   unable to listen on socket: listen tcp :15014: bind: address already in use
```
but these errors are not very important at the moment.

Signed-off-by: Jacek Ewertowski <jewertow@redhat.com>

### CODE REVIEW

File: deploy/src/deployment-servicemesh.yaml [hunk 1]
```diff
@@ -64,6 +64,13 @@ spec:
         olm.relatedImage.v2_2.proxyv2: ${OSSM_PROXY_IMAGE}
         olm.relatedImage.v2_2.wasm-cacher: ${OSSM_PILOT_IMAGE}
         olm.relatedImage.v2_2.rls: ${OSSM_RATELIMIT_IMAGE}
+        # 2.3 images
+        olm.relatedImage.v2_3.cni: ${OSSM_CNI_IMAGE}
+        olm.relatedImage.v2_3.grafana: ${OSSM_GRAFANA_IMAGE}
+        olm.relatedImage.v2_3.pilot: ${OSSM_PILOT_IMAGE}
+        olm.relatedImage.v2_3.prometheus: ${OSSM_PROMETHEUS_IMAGE}
+        olm.relatedImage.v2_3.proxyv2: ${OSSM_PROXY_IMAGE}
+        olm.relatedImage.v2_3.rls: ${OSSM_RATELIMIT_IMAGE}
```
Comment (luksa) [inline deploy/src/deployment-servicemesh.yaml]: I don't know where we set these variables, but using the same variable for both 2.2 and 2.3 is surely wrong. You probably need to change the values for 2.2 to be more like those for 2.1 (with the actual image name + `:${OSSM_22}` at the end) and probably set the `OSSM_22` variable somewhere.

File: manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,794 @@
...
+      [
+        {
+          "apiVersion": "maistra.io/v2",
+          "kind": "ServiceMeshControlPlane",
+          "metadata": {
+            "name": "basic",
+            "namespace": "control-plane-namespace"
+          },
+          "spec": {
+            "version": "v2.1",
```
Comment (luksa) [inline manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml]: This needs to be updated to v2.3 in `build/manifest-templates/clusterserviceversion.yaml`. We forgot to update it in 2.2.

File: manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,794 @@
...
+      kind: ServiceMeshControlPlane
+      displayName: Istio Service Mesh Control Plane
+      description: An Istio control plane installation
+      specDescriptors:
+      - displayName: Control Plane Version
+        description: Specify the version of the control plane you want to install
+        path: version
+        x-descriptors:
+          - 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:General'
+          - 'urn:alm:descriptor:com.tectonic.ui:select:v2.1'
```
Comment (luksa) [inline manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml:694]: Need to add entries for v2.2 and v2.3 above this line. We should also fix this in v2.2.

File: manifests-maistra/maistra.package.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 packageName: maistraoperator
 channels:
 - name: "2.0"
```
Comment (luksa) [inline manifests-maistra/maistra.package.yaml:3]: It's strange that we call the channel `2.0`. Shouldn't it be called `2.x`? @rcernich ?

File: resources/helm/overlays/istio_cni/templates/daemonset.yaml [hunk 5]
```diff
@@ -251,6 +251,64 @@ spec:
...
 {{- end }}
+{{- end }}
+{{- if and (.Values.cni.supportedReleases) (has "v2.3" .Values.cni.supportedReleases) }}
+        - name: install-cni-v2-3
+          image: "{{ .Values.cni.image_v2_3 }}"
+          imagePullPolicy: {{ .Values.cni.imagePullPolicy }}
+          livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8000
```
Comment (luksa) [inline resources/helm/overlays/istio_cni/templates/daemonset.yaml]: Incorrect indentation.

```suggestion
          livenessProbe:
            httpGet:
              path: /healthz
              port: 8000
```

File: resources/helm/v2.2/default/templates/mutatingwebhook.yaml [hunk 6]
```diff
@@ -28,7 +28,7 @@ kind: MutatingWebhookConfiguration
 metadata:
   name: istio-revision-tag-default
   labels:
-    maistra-version: "2.2.1"
+    maistra-version: "2.2.2"
```
Comment (luksa) [inline resources/helm/v2.2/default/templates/mutatingwebhook.yaml]: Please run `make gen` and submit these changes in a separate PR, so that they don't pollute the upgrade to 2.3.

**Review by luksa — COMMENTED** [review]
Found a few small issues. I'd also like this to go in after I add the yaml & helm linters, so that any linting errors are fixed before merging.

I assume you haven't tried deploying a 2.3 SMCP yet. This will uncover additional issues that should also be fixed as part of the rebase process, but we can do it in follow-up PRs.

**Comment by jewertow** [comment]
> I assume you haven't tried deploying a 2.3 SMCP yet.

@luksa as I said in the PR description, I deployed SMCP v2.3 and it worked fine.
I built and pushed required images to my registry, so you can also try to deploy SMCP v2.3 by applying the following diff to `deploy/maistra-operator.yaml`:
```
diff --git a/deploy/maistra-operator.yaml b/deploy/maistra-operator.yaml
index 99f5ece5..fbcff909 100644
--- a/deploy/maistra-operator.yaml
+++ b/deploy/maistra-operator.yaml
@@ -10667,7 +10667,7 @@ spec:

         olm.relatedImage.v1_1.3scale-istio-adapter: quay.io/3scale/3scale-istio-adapter:v1.0.0
         olm.relatedImage.v1_1.citadel: docker.io/maistra/citadel-ubi8:1.1.18
-        olm.relatedImage.v1_1.cni: docker.io/maistra/istio-cni-ubi8:1.1.18
+        olm.relatedImage.v1_1.cni: docker.io/maistra/istio-cni-ubi8:1.1.12
         olm.relatedImage.v1_1.galley: docker.io/maistra/galley-ubi8:1.1.18
         olm.relatedImage.v1_1.grafana: docker.io/maistra/grafana-ubi8:1.1.18
         olm.relatedImage.v1_1.ior: docker.io/maistra/istio-ior-ubi8:1.1.18
@@ -10679,7 +10679,7 @@ spec:
         olm.relatedImage.v1_1.sidecar-injector: docker.io/maistra/sidecar-injector-ubi8:1.1.18

         olm.relatedImage.v2_0.3scale-istio-adapter: quay.io/3scale/3scale-istio-adapter:v2.0.0
-        olm.relatedImage.v2_0.cni: docker.io/maistra/istio-cni-ubi8:2.0.10
+        olm.relatedImage.v2_0.cni: docker.io/maistra/istio-cni-ubi8:2.0.2
         olm.relatedImage.v2_0.grafana: docker.io/maistra/grafana-ubi8:2.0.10
         olm.relatedImage.v2_0.mixer: docker.io/maistra/mixer-ubi8:2.0.10
         olm.relatedImage.v2_0.pilot: docker.io/maistra/pilot-ubi8:2.0.10
@@ -10688,34 +10688,34 @@ spec:
         olm.relatedImage.v2_0.proxyv2: docker.io/maistra/proxyv2-ubi8:2.0.10
         olm.relatedImage.v2_0.wasm-cacher: docker.io/maistra/pilot-ubi8:2.0.10

-        olm.relatedImage.v2_1.cni: quay.io/maistra/istio-cni-ubi8:2.1.4
-        olm.relatedImage.v2_1.grafana: quay.io/maistra/grafana-ubi8:2.1.4
-        olm.relatedImage.v2_1.pilot: quay.io/maistra/pilot-ubi8:2.1.4
-        olm.relatedImage.v2_1.prometheus: quay.io/maistra/prometheus-ubi8:2.1.4
-        olm.relatedImage.v2_1.proxyv2: quay.io/maistra/proxyv2-ubi8:2.1.4
-        olm.relatedImage.v2_1.wasm-cacher: quay.io/maistra/pilot-ubi8:2.1.4
-        olm.relatedImage.v2_1.rls: quay.io/maistra/ratelimit-ubi8:2.1.4
+        olm.relatedImage.v2_1.cni: quay.io/maistra/istio-cni-ubi8:2.1.0
+        olm.relatedImage.v2_1.grafana: quay.io/maistra/grafana-ubi8:2.1.0
+        olm.relatedImage.v2_1.pilot: quay.io/maistra/pilot-ubi8:2.1.0
+        olm.relatedImage.v2_1.prometheus: quay.io/maistra/prometheus-ubi8:2.1.0
+        olm.relatedImage.v2_1.proxyv2: quay.io/maistra/proxyv2-ubi8:2.1.0
+        olm.relatedImage.v2_1.wasm-cacher: quay.io/maistra/pilot-ubi8:2.1.0
+        olm.relatedImage.v2_1.rls: quay.io/maistra/ratelimit-ubi8:2.1.0

-        olm.relatedImage.v2_2.cni: quay.io/maistra/istio-cni-ubi8:2.2.2
-        olm.relatedImage.v2_2.grafana: quay.io/maistra/grafana-ubi8:2.2.2
-        olm.relatedImage.v2_2.pilot: quay.io/maistra/pilot-ubi8:2.2.2
-        olm.relatedImage.v2_2.prometheus: quay.io/maistra/prometheus-ubi8:2.2.2
-        olm.relatedImage.v2_2.proxyv2: quay.io/maistra/proxyv2-ubi8:2.2.2
-        olm.relatedImage.v2_2.wasm-cacher: quay.io/maistra/pilot-ubi8:2.2.2
-        olm.relatedImage.v2_2.rls: quay.io/maistra/ratelimit-ubi8:2.2.2
+        olm.relatedImage.v2_2.cni: quay.io/jewertow/istio-cni:2.2.2
+        olm.relatedImage.v2_2.grafana: quay.io/maistra/grafana-ubi8:2.1.0
+        olm.relatedImage.v2_2.pilot: quay.io/jewertow/pilot:2.2.2
+        olm.relatedImage.v2_2.prometheus: quay.io/maistra/prometheus-ubi8:2.1.0
+        olm.relatedImage.v2_2.proxyv2: quay.io/jewertow/proxyv2:2.2.2
+        olm.relatedImage.v2_2.wasm-cacher: quay.io/jewertow/pilot:2.2.2
+        olm.relatedImage.v2_2.rls: quay.io/maistra/ratelimit-ubi8:2.1.0

-        olm.relatedImage.v2_3.cni: quay.io/maistra/istio-cni-ubi8:2.3.0
-        olm.relatedImage.v2_3.grafana: quay.io/maistra/grafana-ubi8:2.3.0
-        olm.relatedImage.v2_3.pilot: quay.io/maistra/pilot-ubi8:2.3.0
-        olm.relatedImage.v2_3.prometheus: quay.io/maistra/prometheus-ubi8:2.3.0
-        olm.relatedImage.v2_3.proxyv2: quay.io/maistra/proxyv2-ubi8:2.3.0
-        olm.relatedImage.v2_3.rls: quay.io/maistra/ratelimit-ubi8:2.3.0
+        olm.relatedImage.v2_3.cni: quay.io/jewertow/istio-cni:2.3.0
+        olm.relatedImage.v2_3.grafana: quay.io/maistra/grafana-ubi8:2.1.0
+        olm.relatedImage.v2_3.pilot: quay.io/jewertow/pilot:2.3.0
+        olm.relatedImage.v2_3.prometheus: quay.io/maistra/prometheus-ubi8:2.1.0
+        olm.relatedImage.v2_3.proxyv2: quay.io/jewertow/proxyv2:2.3.0
+        olm.relatedImage.v2_3.rls: quay.io/maistra/ratelimit-ubi8:2.1.0

     spec:
       serviceAccountName: istio-operator
       containers:
       - name: istio-operator
-        image: quay.io/maistra/istio-ubi8-operator:2.3.0
+        image: quay.io/jewertow/istio-operator:2.3.0
         ports:
         - containerPort: 11999
           name: validation

```

File: resources/helm/overlays/istio_cni/templates/daemonset.yaml (hunk 5 above)
Comment (jewertow) [reply resources/helm/overlays/istio_cni/templates/daemonset.yaml]: Thanks, I will fix it.

**Comment by luksa** [comment]
Oh, sorry. I assumed that istio-cni wouldn't work because of the bad indentation in the DaemonSet YAML, which caused me to assume that you haven't yet tried deploying the operator and bookinfo. If bookinfo works, that's great!

File: deploy/src/deployment-servicemesh.yaml (hunk 1 above)
Comment (jewertow) [reply deploy/src/deployment-servicemesh.yaml]: These variables are set in CPaaS, but you're right that these variables cannot be the same as for 2.2.

@jwendell do you have any suggestion? Can I name these variables to `OSSM_CNI_2_3_*`?

File: resources/helm/v2.2/default/templates/mutatingwebhook.yaml (hunk 6 above)
Comment (jewertow) [reply resources/helm/v2.2/default/templates/mutatingwebhook.yaml]: Good idea, I will do it.

File: deploy/src/deployment-servicemesh.yaml (hunk 1 above)
Comment (luksa) [reply deploy/src/deployment-servicemesh.yaml]: I think you need to keep the current name, and instead change the 2.2 values. For example:
```
olm.relatedImage.v2_2.rls: ${OSSM_RATELIMIT_IMAGE}
```
should be changed to:
```
olm.relatedImage.v2_2.rls: registry-proxy.engineering.redhat.com/rh-osbs/openshift-service-mesh-ratelimit-rhel8:${OSSM_22}
```

File: manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml (hunk 2 above)
Comment (jewertow) [reply manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml]: Done

File: manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml (hunk 3 above)
Comment (jewertow) [reply manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml:694]: Done

File: deploy/src/deployment-servicemesh.yaml (hunk 1 above)
Comment (jewertow) [reply deploy/src/deployment-servicemesh.yaml]: Done

File: Makefile [hunk 7]
```diff
@@ -14,8 +14,8 @@
 
 -include Makefile.overrides
 
-MAISTRA_VERSION        ?= 2.2.2
-MAISTRA_BRANCH         ?= maistra-2.2
+MAISTRA_VERSION        ?= 2.3.0
+MAISTRA_BRANCH         ?= maistra-2.3
 REPLACES_PRODUCT_CSV   ?= 2.2.1
 REPLACES_COMMUNITY_CSV ?= 2.2.1
```
Comment (luksa) [inline Makefile]: I think this needs to be updated, since we've released 2.2.2.

File: Makefile [hunk 8]
```diff
@@ -234,7 +265,7 @@ ifneq "${OSSM_MANIFEST_DATE}" ""
 endif
 
 .PHONY: update-charts
-update-charts: update-1.1-charts update-2.0-charts update-2.1-charts update-2.2-charts
+update-charts: update-1.1-charts update-2.0-charts update-2.1-charts update-2.2-charts update-2.3-charts
 
 .PHONY: update-templates
 update-templates: update-1.1-templates update-2.0-templates update-2.1-templates
```
Comment (luksa) [inline Makefile]: I think we need to add `update-2.2-templates` here.

File: Makefile (hunk 7 above)
Comment (jewertow) [reply Makefile]: Done

File: Makefile (hunk 8 above)
Comment (jewertow) [reply Makefile]: Done

**Review by luksa — APPROVED** [review]

**Comment by luksa** [comment]
@jewertow great work!

**Comment by jewertow** [comment]
Thanks
